File: dmn-signavio/src/test/resources/dmn2java/exported/proto/expected/proto3/java/dmn/example-credit-decision/type/Applicant.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "applicant"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(as = type.ApplicantImpl.class)
public interface Applicant extends com.gs.dmn.runtime.DMNType {
    static Applicant toApplicant(Object other) {
        if (other == null) {
            return null;
        } else if (Applicant.class.isAssignableFrom(other.getClass())) {
            return (Applicant)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            ApplicantImpl result_ = new ApplicantImpl();
            result_.setName((String)((com.gs.dmn.runtime.Context)other).get("name", "Name"));
            result_.setAge((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("age", "Age"));
            result_.setCreditScore((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("creditScore", "Credit score"));
            result_.setPriorIssues((List<String>)((com.gs.dmn.runtime.Context)other).get("priorIssues", "Prior issues"));
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toApplicant(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else if (other instanceof proto.Applicant) {
            ApplicantImpl result_ = new ApplicantImpl();
            result_.setName(((proto.Applicant) other).getName());
            result_.setAge(java.math.BigDecimal.valueOf(((proto.Applicant) other).getAge()));
            result_.setCreditScore(java.math.BigDecimal.valueOf(((proto.Applicant) other).getCreditScore()));
            result_.setPriorIssues(((List<String>) (((proto.Applicant) other).getPriorIssuesList() == null ? null : ((proto.Applicant) other).getPriorIssuesList().stream().map(e -> e).collect(java.util.stream.Collectors.toList()))));
            return result_;
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), Applicant.class.getSimpleName()));
        }
    }

    static proto.Applicant toProto(Applicant other) {
        proto.Applicant.Builder result_ = proto.Applicant.newBuilder();
        if (other != null) {
            String nameProto_ = (((Applicant) other).getName() == null ? "" : ((Applicant) other).getName());
            result_.setName(nameProto_);
            Double ageProto_ = (((Applicant) other).getAge() == null ? 0.0 : ((Applicant) other).getAge().doubleValue());
            result_.setAge(ageProto_);
            Double creditScoreProto_ = (((Applicant) other).getCreditScore() == null ? 0.0 : ((Applicant) other).getCreditScore().doubleValue());
            result_.setCreditScore(creditScoreProto_);
            List<String> priorIssuesProto_ = ((List) (((Applicant) other).getPriorIssues() == null ? null : ((Applicant) other).getPriorIssues().stream().map(e -> (e == null ? "" : e)).collect(java.util.stream.Collectors.toList())));
            if (priorIssuesProto_ != null) {
                result_.addAllPriorIssues(priorIssuesProto_);
            }
        }
        return result_.build();
    }

    static List<proto.Applicant> toProto(List<Applicant> other) {
        if (other == null) {
            return null;
        } else {
            return other.stream().map(o -> toProto(o)).collect(java.util.stream.Collectors.toList());
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Name")
    String getName();

    @com.fasterxml.jackson.annotation.JsonGetter("Age")
    java.math.BigDecimal getAge();

    @com.fasterxml.jackson.annotation.JsonGetter("Credit score")
    java.math.BigDecimal getCreditScore();

    @com.fasterxml.jackson.annotation.JsonGetter("Prior issues")
    List<String> getPriorIssues();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("name", getName());
        context.put("age", getAge());
        context.put("creditScore", getCreditScore());
        context.put("priorIssues", getPriorIssues());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Applicant other = (Applicant) o;
        if (this.getAge() != null ? !this.getAge().equals(other.getAge()) : other.getAge() != null) return false;
        if (this.getCreditScore() != null ? !this.getCreditScore().equals(other.getCreditScore()) : other.getCreditScore() != null) return false;
        if (this.getName() != null ? !this.getName().equals(other.getName()) : other.getName() != null) return false;
        if (this.getPriorIssues() != null ? !this.getPriorIssues().equals(other.getPriorIssues()) : other.getPriorIssues() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getAge() != null ? this.getAge().hashCode() : 0);
        result = 31 * result + (this.getCreditScore() != null ? this.getCreditScore().hashCode() : 0);
        result = 31 * result + (this.getName() != null ? this.getName().hashCode() : 0);
        result = 31 * result + (this.getPriorIssues() != null ? this.getPriorIssues().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("Age=" + getAge());
        result_.append(", Credit score=" + getCreditScore());
        result_.append(", Name=" + getName());
        result_.append(", Prior issues=" + getPriorIssues());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-test-cases/signavio/dmn/dmn2java/expected/proto/proto3/kotlin/dmn/example-credit-decision/type/Applicant.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "applicant"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.ApplicantImpl::class)
interface Applicant : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("Name")
    val name: String?

    @get:com.fasterxml.jackson.annotation.JsonGetter("Age")
    val age: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("Credit score")
    val creditScore: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("Prior issues")
    val priorIssues: List<String?>?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("name", this.name)
        context.put("age", this.age)
        context.put("creditScore", this.creditScore)
        context.put("priorIssues", this.priorIssues)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as Applicant
        if (if (this.age != null) this.age != other.age else other.age != null) return false
        if (if (this.creditScore != null) this.creditScore != other.creditScore else other.creditScore != null) return false
        if (if (this.name != null) this.name != other.name else other.name != null) return false
        if (if (this.priorIssues != null) this.priorIssues != other.priorIssues else other.priorIssues != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.age != null) this.age.hashCode() else 0)
        result = 31 * result + (if (this.creditScore != null) this.creditScore.hashCode() else 0)
        result = 31 * result + (if (this.name != null) this.name.hashCode() else 0)
        result = 31 * result + (if (this.priorIssues != null) this.priorIssues.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("Age=" + age)
        result_.append(", Credit score=" + creditScore)
        result_.append(", Name=" + name)
        result_.append(", Prior issues=" + priorIssues)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toApplicant(other: Any?): Applicant? {
            if (other == null) {
                return null
            } else if (other is Applicant?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = ApplicantImpl()
                result_.name = other.get("name", "Name") as String?
                result_.age = other.get("age", "Age") as java.math.BigDecimal?
                result_.creditScore = other.get("creditScore", "Credit score") as java.math.BigDecimal?
                result_.priorIssues = other.get("priorIssues", "Prior issues") as List<String?>?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toApplicant(other.toContext())
            } else if (other is proto.Applicant) {
                var result_: ApplicantImpl = ApplicantImpl()
                result_.name = (other as proto.Applicant).getName()
                result_.age = java.math.BigDecimal.valueOf((other as proto.Applicant).getAge())
                result_.creditScore = java.math.BigDecimal.valueOf((other as proto.Applicant).getCreditScore())
                result_.priorIssues = ((other as proto.Applicant).getPriorIssuesList()?.stream()?.map({e -> e})?.collect(java.util.stream.Collectors.toList()) as List<String?>?)
                return result_
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), Applicant::class.java.getSimpleName()))
            }
        }

        @JvmStatic
        fun toProto(other: Applicant?): proto.Applicant {
            var result_: proto.Applicant.Builder = proto.Applicant.newBuilder()
            if (other != null) {
                var nameProto_: String = (if ((other as Applicant).name == null) "" else (other as Applicant).name!!)
                result_.setName(nameProto_)
                var ageProto_: Double = (if ((other as Applicant).age == null) 0.0 else (other as Applicant).age!!.toDouble())
                result_.setAge(ageProto_)
                var creditScoreProto_: Double = (if ((other as Applicant).creditScore == null) 0.0 else (other as Applicant).creditScore!!.toDouble())
                result_.setCreditScore(creditScoreProto_)
                var priorIssuesProto_: List<String?>? = (other as Applicant).priorIssues?.stream()?.map({e -> (if (e == null) "" else e!!)})?.collect(java.util.stream.Collectors.toList())
                if (priorIssuesProto_ != null) {
                    result_.addAllPriorIssues(priorIssuesProto_)
                }
            }
            return result_.build()
        }

        @JvmStatic
        fun toProto(other: List<Applicant?>?): List<proto.Applicant>? {
            if (other == null) {
                return null
            } else {
                return other.stream().map({o -> toProto(o)}).collect(java.util.stream.Collectors.toList())
            }
        }
    }
}
